### ISSUE TITLE
`dotnet package search` should expose underlying errors when `detailed` verbosity is set

### ISSUE THREAD

**Issue by julealgon** [NuGet/Home#13338]
### NuGet Product(s) Affected

dotnet.exe

### Current Behavior

When running `dotnet package search` and getting an error, the underlying exception is never surfaced even when passing the `--verbosity detailed` setting.

### Desired Behavior

I believe exposing the underlying error, at the very least in higher verbosity modes, would help with debugging issues that are otherwise hard to understand.

### Additional Context

I found out about this while attempting to find a solution to my problem in:
- https://github.com/NuGet/Home/issues/13331

I could only see the actual error when calling the API directly via Postman.

**Comment by jeffkl** [comment]
@julealgon can you provide more info as to what errors you are expecting to see?  NuGet logs errors as a different kind of message so they won't be affected by verbosity.  It sounds like NuGet isn't logging errors in your case?  We'll need more info about errors that aren't being logged.

**Comment by julealgon** [comment]
> @julealgon can you provide more info as to what errors you are expecting to see? 

As with my previous issue, linked there, I expected to see a variation of this:

> Source: GithubFeed (https://nuget.pkg.github.com/our-company/index.json)
warn : Failed to retrieve metadata from source 'https://nuget.pkg.github.com/our-company/query?q=OurCompany&skip=0&take=20&prerelease=false&semVerLevel=2.0.0'.
HTTP 404. "error": "The expected resource was not found."

Instead of this, which is what is returned today (no underlying error information):
> Source: GithubFeed (https://nuget.pkg.github.com/our-company/index.json)
warn : Failed to retrieve metadata from source 'https://nuget.pkg.github.com/our-company/query?q=OurCompany&skip=0&take=20&prerelease=false&semVerLevel=2.0.0'.

> NuGet logs errors as a different kind of message so they won't be affected by verbosity. It sounds like NuGet isn't logging errors in your case? We'll need more info about errors that aren't being logged.

I'm not using `NuGet.exe` directly. I don't know how `dotnet` CLI interacts with `NuGet.exe`.

**Comment by nkolev92** [comment]
@Nigusu-Allehu 

Can you please take a quick look at this and see if we're "hiding", or "swallowing" any failure details. 

fwiw, @julealgon messages like `Failed to retrieve metadata from source 'https://nuget.pkg.github.com/our-company/query?q=OurCompany&skip=0&take=20&prerelease=false&semVerLevel=2.0.0'` are all the information we have sometimes, it's what restore would warn about as well.

**Comment by julealgon** [comment]
> fwiw, @julealgon messages like `Failed to retrieve metadata from source 'https://nuget.pkg.github.com/our-company/query?q=OurCompany&skip=0&take=20&prerelease=false&semVerLevel=2.0.0'` are all the information we have sometimes, it's what restore would warn about as well.

I don't believe that's the case here @nkolev92 . In my scenario, there was clearly an HTTP status code with an accompanying error message in the underlying request which would've helped debug the issue from the CLI.

It seems to me that the code here (in the CLI tool) is just checking for whether or not the request was "successful", and then exposing a high level error for the current operation it is doing, while not propagating the underlying HTTP status and the response payload forward.
